modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala

```diff
@@ -256,6 +256,91 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
+  test("Resource embedding in Scala Native") {
+    val projectDir       = "nativeres"
+    val resourceContent  = "resource contents"
+    val resourceFileName = "embeddedfile.txt"
+    val inputs = TestInputs(
+      os.rel / projectDir / "main.scala" ->
+        s"""|//> using platform "scala-native"
+            |//> using resourceDir "resources"
+            |
+            |import java.nio.charset.StandardCharsets
+            |import java.io.{BufferedReader, InputStreamReader}
+            |import java.io.FileInputStream
+            |
+            |object Main {
+            |  def main(args: Array[String]): Unit = {
+            |    val inputStream = getClass().getResourceAsStream("/$resourceFileName")
+            |    val nativeResourceText = new BufferedReader(
+            |      new InputStreamReader(inputStream, StandardCharsets.UTF_8)
+            |    ).readLine()
+            |    println(nativeResourceText)
+            |  }
+            |}
+            |""".stripMargin,
+      os.rel / projectDir / "resources" / resourceFileName -> resourceContent
+    )
+    inputs.fromRoot { root =>
+      val output =
+        os.proc(TestUtil.cli, extraOptions, projectDir, "-q")
+          .call(cwd = root)
+          .out.text().trim
+      println(output)
+      expect(output == resourceContent)
+    }
+  }
+
+  test("Scala Native C Files are correctly handled as a regular Input") {
+    val projectDir      = "native-interop"
+    val interopFileName = "bindings.c"
+    val interopMsg      = "Hello C!"
+    val inputs = TestInputs(
+      os.rel / projectDir / "main.scala" ->
+        s"""|//> using platform "scala-native"
+            |
+            |import scala.scalanative.unsafe._
+            |
+            |@extern
+            |object Bindings {
+            |  @name("scalanative_print")
+            |  def print(): Unit = extern
+            |}
+            |
+            |object Main {
+            |  def main(args: Array[String]): Unit = {
+            |    Bindings.print()
+            |  }
+            |}
+            |""".stripMargin,
+      os.rel / projectDir / interopFileName ->
+        s"""|#include <stdio.h>
+            |
+            |void scalanative_print() {
+            |    printf("$interopMsg\\n");
+            |}
+            |""".stripMargin
+    )
+    inputs.fromRoot { root =>
+      val output =
+        os.proc(TestUtil.cli, extraOptions, projectDir, "-q")
+          .call(cwd = root)
+          .out.text().trim
+      expect(output == interopMsg)
+
+      os.move(root / projectDir / interopFileName, root / projectDir / "bindings2.c")
+      val output2 =
+        os.proc(TestUtil.cli, extraOptions, projectDir, "-q")
+          .call(cwd = root)
+          .out.text().trim
+
+      // LLVM throws linking errors if scalanative_print is internally repeated.
```

Comment:
Why does `copyResourcesToDirWithMapping` get "internally repeated" when renaming `bindings.c`? The exact name of the C file (apart from the `.c` extension) shouldn't matter, does it?

Reply:
It would previously, since the binding.c and the renamed binding.c files would both appear in .scala-build (the original file would not be deleted without the mapping feature), leading to repeated symbols which in turn would give us linking errors. This test checks whether that still happens or not. I'll adjust the phrasing of the comment to make it more clear